topics/transcriptomics/tutorials/droplet-quantification-preprocessing/tutorial.md

```diff
@@ -0,0 +1,752 @@
+> {: .solution}
+{: .question}
+
+
+## Generate a transcript to gene map
+
+Gene-level, rather than transcript-level, quantification is standard in scRNA-seq, which means that that the expression level of alternatively spliced RNA molecules are combined to create gene-level values. Droplet-based scRNA-seq techniques only sample one end each transcript, so lack the full-molecule coverage that would be required to accurately quantify different transcript isoforms.
+
+To generate gene-level quantifications based on transcriptome quantification, Alevin and similar tools require a conversion between transcript and gene identifiers. We can derive a transcript-gene conversion from the gene annotations available in genome resources such as Ensembl. The transcripts in such a list need to match the ones we will use later to build a binary transcriptome index. If you were using spike-ins, you'd need to add these to the transcriptome and the transcript-gene mapping.
+
+In your example data you will see the murine reference annotation as retrieved from Ensembl in GTF format. This annotation contains gene, exon, transcript and all sorts of other information on the sequences. We will use these to generate the transcript/ gene mapping by passing that information to a tool that extracts just the transcript identifiers we need.
+
+> ### {% icon question %} Questions
+>
+> Which of the 'attributes' in the last column of the GTF files contains the transcript and gene identifiers?
+>
+>
+>   > ### {% icon tip %} Hint
+>   >
+>   > The file is organised such that the last column (headed 'Group') contains a wealth of information in the format: attribute1 "information associated with attribute 1";attribute2 "information associated with attribute 2" etc.
+>   {: .tip}
+>
+> > ### {% icon solution %} Solution
+> > *gene_id* and *transcript_id* are each followed by "ensembl gene_id" and "ensembl transcript_id"
+> {: .solution}
+{: .question}
+
+It's now time to parse the GTF file using the [rtracklayer](https://bioconductor.org/packages/release/bioc/html/rtracklayer.html) package in R. This parsing will give us a conversion table with a list of transcript identifiers and their corresponding gene identifiers for counting. Additionally, because we will be generating our own binary index (more later!), we also need to input our FASTA so that it can be filtered to only contain transcriptome information found in the GTF.
+
+> ### {% icon hands_on %} Hands-on: Generate a filtered FASTA and transcript-gene map
+>
+> 1. {% tool [GTF2GeneList](https://humancellatlas.usegalaxy.eu/root?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/gtf2gene_list/_ensembl_gtf2gene_list/1.42.1+galaxy6) %} with the following parameters:
```

Comment:
Could you strip off the leading URL like below for the tool links, then it will work on whichever Galaxy the user is on (if they use the GTN-in-Galaxy hat icon thingy)

```suggestion
> 1. {% tool [GTF2GeneList](toolshed.g2.bx.psu.edu/repos/ebi-gxa/gtf2gene_list/_ensembl_gtf2gene_list/1.42.1+galaxy6) %} with the following parameters:
```